library/src/com/google/maps/android/kml/KmlContainer.java

```diff
@@ -0,0 +1,254 @@
+package com.google.maps.android.kml;
+
+import com.google.android.gms.maps.model.GroundOverlay;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+/**
+ * Represents a KML Document or Folder.
+ */
+public class KmlContainer {
+
+    private final HashMap<String, String> mProperties;
+
+    private final HashMap<KmlPlacemark, Object> mPlacemarks;
+
+    private final ArrayList<KmlContainer> mContainers;
+
+    private final HashMap<KmlGroundOverlay, GroundOverlay> mGroundOverlays;
+
+    private final HashMap<String, String> mStyleMap;
+
+    private HashMap<String, KmlStyle> mStyles;
+
+    private String mContainerId;
+
+    public KmlContainer() {
+        mProperties = new HashMap<String, String>();
+        mPlacemarks = new HashMap<KmlPlacemark, Object>();
+        mStyles = new HashMap<String, KmlStyle>();
+        mStyleMap = new HashMap<String, String>();
+        mContainers = new ArrayList<KmlContainer>();
+        mGroundOverlays = new HashMap<KmlGroundOverlay, GroundOverlay>();
+        mContainerId = null;
+    }
+
+    /**
+     * Gets the Container ID if it is specified
+     *
+     * @return Container ID or null if not set
+     */
+    public String getContainerId() {
+        return mContainerId;
+    }
+
+    /**
+     * Assigns a string value to the Contianer
+     *
+     * @param containerId   String value to assign to the contianer
+     */
+    public void setContainerId(String containerId) {
+        mContainerId = containerId;
+    }
+
+    /*package*/ ArrayList<KmlContainer> getNestedContainers() {
+        return mContainers;
+    }
+
+    /**
+     * Gets a style based on an ID
+     */
+    public KmlStyle getStyle(String styleID) {
+        return mStyles.get(styleID);
+    }
+
+    /**
+     * @return HashMap of styles, with key values representing style name (ie, color) and
+     * value representing style value (ie #FFFFFF)
+     */
+    /* package */ HashMap<String, KmlStyle> getStyles() {
+        return mStyles;
+    }
+
+    /**
+     * Takes an ArrayList of styles and assigns these folders with the styles
+     */
+    /* package */ void setStyles(HashMap<String, KmlStyle> styles) {
+        mStyles = styles;
+    }
+
+    /**
+     * @param placemarks Placemark for the container to contain
+     * @param object     Corresponding GoogleMap map object of the basic_placemark (if it has been
+     *                   added
+     *                   to the map)
```

Comment:
Too many line breaks in the '@param object' javadoc. And you've specified in the docs that it's a GoogleMap object, but the type here is Object. Why are they different?